Gems/Atom/RHI/Code/Include/Atom/RHI/Object.h

```diff
@@ -60,5 +62,27 @@ namespace AZ::RHI
         virtual void SetNameInternal(const AZStd::string_view& name);
 
         Name m_name;
+
+        // SetName()/GetName() can be called from Multiple Threads, the Name class
+        // is not thread safe. Typical multi threaded case comes from MeshDrawPackets::DoUpdate()
+        // which occurs across multiple jobs. Without this mutex you may experience random crashes,
+        // in particular for large levels with lots of meshes, like SinglePlayer level from StarterGame project.
+        // TODO: Revisit this? Maybe m_name should be AZStd::string instead of Name.
+        // Maybe this mutex should be unique per object, but increases the memory footprint of each RHI Object
+        // by 40 to 80Bytes.
+        // So far for large scenes like StarterGame, making this mutex a unique instance per Object
+        // did not appear to be perceivable as more performant.
+#define RHI_OBJECT_STATIC_MUTEX (0)
+#if RHI_OBJECT_STATIC_MUTEX
+        // Vulkan: It takes 3m:00s to enter game mode with SinglePlayer level from StarterGame project.
+        // Vulkan: It takes 4m:00s minutes to exit game mode with SinglePlayer level from StarterGame project.
+        static AZStd::recursive_mutex m_nameMutex;
+#else
+        // Vulkan: It takes 1m:50s to enter game mode with SinglePlayer level from StarterGame project.
+        // Vulkan: It takes 2m:46s to exit game mode with SinglePlayer level from StarterGame project.
+        // DX12: It takes 0m:23s to enter game mode with SinglePlayer level from StarterGame project.
+        // DX12: It takes 0m:26s to exit game mode with SinglePlayer level from StarterGame project.
```

Comment:
Why are there two set of stats here? What's the difference between them?

Reply:
It shows the performance difference between Vulkan and DX12 when it comes to Loading/Unloading Large levels.
It seems the DX12 RHI doesn't use the SetName()/GetName() APIs?

Reply:
I didn't read the comment properly :( I thought all the lines said 'enter' the game mode. I feel like this change is a bit worrisome and would like to see if there is a better solution here.